### PR TITLE
[NSA-9630] add create role based service via UI in hidden state

### PR DESCRIPTION
**Pull request by alexplattdfe**
- Add ability for role based service to 'hide from support' and 'hide from approvers' when creating a service

### CODE REVIEW

File: src/app/services/views/chooseServiceType.ejs [hunk 1]
```diff
@@ -85,6 +85,24 @@
...
+                    Hides <span class="govuk-!-font-weight-bold">role-based</span> services from the support console.
+                  </div>
+                </div>
+                <div class="govuk-checkboxes__item">
+                  <input class="govuk-checkboxes__input" id="hideApprover" name="hideApprover" type="checkbox" value="hideApprover" <%= (locals.hideApprover) ? 'checked' : '' %>>
+                  <label class="govuk-label govuk-checkboxes__label" for="hideApprover">
+                    Hide from approvers
+                  </label>
+                  <div id="hideApprover-item-hint" class="govuk-hint govuk-checkboxes__hint">
+                    Hides <span class="govuk-!-font-weight-bold">role-based</span> services from users so they cannot request the service, and from approvers so they cannot add the service to their account or users in their orgnisation.
```
Comment (AMcCartneyMTH) [inline src/app/services/views/chooseServiceType.ejs]: Typo "orgnisation" should be "organisation"

File: test/app/services/postConfirmNewService.test.js [hunk 2]
```diff
@@ -505,4 +508,60 @@ describe("when displaying the post create new service", () => {
...
+    await postConfirmNewService(req, res);
+
+    expect(createServiceRaw.mock.calls).toHaveLength(1);
+    expect(
+      createServiceRaw.mock.calls[0][0].relyingParty.params.hideSupport,
+    ).toBe(true);
+    expect(res.redirect.mock.calls).toHaveLength(1);
+    expect(res.redirect.mock.calls[0][0]).toBe("/users");
+  });
 });
```
Comment (AMcCartneyMTH) [inline test/app/services/postConfirmNewService.test.js:931]: We could cover the the 5 scenarios from the ticket explicitly?

Scenario 1 - Onboarding an ID-only service marks it as hidden from users as well as support, even if the new options aren’t selected
Scenario 2 - Onboarding a role-based service with none of the hide options selected doesn’t hide the service
Scenario 3 - Onboarding a role-based service with the hide from users/approvers option selected will hide the service
Scenario 4 - Onboarding a role-based service with the hide from support console option selected will hide the service
Scenario 5 - Onboarding a role-based service with the hide from users/approvers, hide from support console, and hide from contact us options selected will hide the service

Comment (AMcCartneyMTH) [inline test/app/services/postConfirmNewService.test.js:931]: Consider these edge case tests:

It should have all four flags set when all are provided for standardServiceType
It should have isHiddenService and hideSupport true when hideFromUserServices and hideSupport are set
It shoulld have helpHidden and hideApprover true when both hideFromContactUs and hideApprover are set
It should have helpHidden true when hideFromContactUs is set for standardServiceType

**Review by AMcCartneyMTH — CHANGES_REQUESTED** [review]
Just a typo on one of the views and some additional tests.

**Review by AMcCartneyMTH — APPROVED** [review]

**Review by shravani-kondaparti — APPROVED** [review]
